src/psyclone/parse/kernel.py

```diff
@@ -534,40 +534,49 @@ def get_procedure(ast, name, modast):
                             "Kernel type {0} binds to a specific procedure but"
                             " does not use 'code' as the generic name.".
                             format(name))
-                    bname = statement.bname
+                    bname = statement.bname.lower()
```

Comment:
On reading the docstring I've realised that this routine returns the parse tree for "the subroutine statement". That's OK when we have a single type-bound procedure but it's not clear when we have multiple implementations referenced from an interface. Please check how this information is used in the caller and update appropriately - you may need to modify how `code` is set below too.
Please could you also the docstring descriptions of the error descriptions to mention interfaces where appropriate.

Reply:
Well this is a bit complicated as when the constructor is called on kernelProcedure, it calls get_procedure and the code block is then an internal data member. Its not clear how the code block is used - if at all because all the code generated code seems to be correct and the tests all pass whether I have an interface with several module procedures, or just one type bound procedure. Perhaps we need to discuss this?

Reply:
If there is more than one Module procedure in the interface, then it no longer returns the subroutine code block as there would have to be several blocks. Just need to fix up the inlining.

Reply:
As discussed, added a check for no code to inlining and a test for this.